spec/requests/api/v1.2/order_processes_controller_spec.rb

```diff
@@ -391,6 +391,43 @@
     end
   end
 
+  describe "GET /order_processes/:id/tags #tags" do
+    let(:bad_order_process_id) { order_process.id + 1 }
+
+    before do
+      allow(Insights::API::Common::RBAC::Access).to receive(:new).and_return(catalog_access)
+      allow(catalog_access).to receive(:process).and_return(catalog_access)
+      order_process.tag_add("yay_tag")
+    end
+
+    context "when listing all tags for an order process" do
+      let(:catalog_access) { instance_double(Insights::API::Common::RBAC::Access, :scopes => %w[admin]) }
+
+      it "returns the tags for the order process" do
+        get "#{api_version}/order_processes/#{order_process.id}/tags", :headers => default_headers
+
+        expect(json["meta"]["count"]).to eq 1
+        expect(json["data"].first["tag"]).to eq Tag.new(:name => "yay_tag").to_tag_string
+      end
+
+      it "returns not found when order process missing" do
+        get "#{api_version}/order_processes/#{bad_order_process_id}/tags", :headers => default_headers
+
+        expect(response).to have_http_status(404)
+      end
+    end
+
+    context "when listing all tags for an order process you do not have access to" do
+      let(:catalog_access) { instance_double(Insights::API::Common::RBAC::Access, :scopes => %w[group]) }
+
+      it "returns 403" do
+        get "#{api_version}/order_processes/#{order_process.id}/tags", :headers => default_headers
+
+        expect(response).to have_http_status(403)
+      end
+    end
+  end
```

Comment:
I think this should probably be moved to a new `spec/requests/api/v1.2/tags_controller_spec.rb` instead. You will have to add two other contexts for the `/portfolios/:id/tag` and `/portfolio_items/:id/tag` endpoints, but they're already in the `spec/requests/api/v1.0/tags_spec.rb` so they can be moved over.

This feels like a piece of code we can somehow extract to make better and more robust, instead of always having to add an `elsif` when we introduce new objects that have tags, but I don't have a very good initial thought on what that looks like. Something to look at to refactor down the line.

Reply:
Yeah, now I am thinking about how to refactor this to remove `elsif`